spec/models/question_mover_spec.rb

```diff
@@ -0,0 +1,316 @@
+            let(:relation) { :after }
+            it_behaves_like "succeeds"
+          end
+
+          context "within existing division depth block inside other division block" do
+            let(:current_division) { div_c1 }
+            let(:question) { q41 }
+            let(:target) { q102 }
+            let(:relation) { :after }
+            it_behaves_like "succeeds"
+          end
+
+          context "before existing division depth block with :inside" do
+            let(:current_division) { div_c1 }
+            let(:question) { q41 }
+            let(:target) { q10 }
+            let(:relation) { :inside }
+            let(:error) { "must be adjacent to questions of same division depth" }
+            it_behaves_like "denies"
+          end
+
+          context "before existing division depth block with :before" do
+            let(:current_division) { div_c1 }
+            let(:question) { q41 }
+            let(:target) { q100 }
+            let(:relation) { :before }
+            let(:error) { "must be adjacent to questions of same division depth" }
+            it_behaves_like "denies"
+          end
+
+          context "after existing division depth block with :before" do
+            let(:current_division) { div_c1 }
+            let(:question) { q113 }
+            let(:target) { q105 }
+            let(:relation) { :before }
+            let(:error) { "must be adjacent to questions of same division depth" }
+            it_behaves_like "denies"
+          end
+
+          context "after existing division depth block with :after" do
+            let(:current_division) { div_c1 }
+            let(:question) { q113 }
+            let(:target) { q105 }
+            let(:relation) { :after }
+            let(:error) { "must be adjacent to questions of same division depth" }
+            it_behaves_like "denies"
+          end
+        end
+
+        context "when no questions of same division depth" do
+          context "at end of group" do
+            let(:current_division) { div_d }
+            let(:question) { q118 }
+            let(:target) { q11 }
+            let(:relation) { :after }
+            it_behaves_like "succeeds"
+          end
+
+          context "not at end of group" do
+            let(:current_division) { div_d }
+            let(:question) { q118 }
+            let(:target) { q11 }
+            let(:relation) { :before }
+            let(:error) { "must be adjacent to questions of same division depth" }
+            it_behaves_like "denies"
+          end
+        end
+
+        context "when group is empty" do
+          context "at end of group" do
+            let(:current_division) { div_d }
+            let(:question) { q118 }
+            let(:target) { q7 }
+            let(:relation) { :inside }
+            it_behaves_like "succeeds"
+          end
+        end
+      end
+    end
+  end
+
+  context "moving question with ancestor division than current division" do
```

Comment:
is there a word missing here? with ancestor div different than curr div?

Reply:
Ah yes, I'll fix.